### DISCUSSION TITLE
Python - Ultralytics YOLOv8 Docs

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#4141] [Docs]
# Python - Ultralytics YOLOv8 Docs

Boost your Python projects with object detection, segmentation and classification using YOLOv8. Explore how to load, train, validate, predict, export, track and benchmark models with ease.

https://docs.ultralytics.com/usage/python/

**Comment by yhl668899** [comment]
when i use "Results usage"，
result.masks.xy        # x,y segments (pixels), List[segment] * N
debug this result in pycharm, why i got '\n' at the end of each [points]？

**Reply by pderrenger** [reply]
@yhl668899 the `\n` character you're seeing at the end of each list of points when debugging in PyCharm is likely a visual representation of the newline character that's used to separate items when printing lists or arrays for readability. This is a common feature in many IDEs and text editors to make the output more human-readable.

When you print or view lists in a debug console, each item is often placed on a new line, and `\n` represents the end of one line and the start of another. This doesn't affect the actual data structure; it's just a way to display the data. The points in `result.masks.xy` are still stored in the list as you would expect, without any newline characters. If you were to process or iterate over the list programmatically, you wouldn't encounter these newline characters as part of the data. They are purely for display purposes in the debug environment.

**Comment by TimDev-r** [comment]
Remark suggestion:
In Usage->Python->Export to TensorRT chapter shouldn't it be:
  model.export(format='engine', device=0) instead of
  model.export(format='onnx', device=0) ?

**Reply by pderrenger** [reply]
@TimDev-r thank you for your suggestion! You're correct, when exporting a model specifically for TensorRT, the format should be specified as `'engine'`. The correct usage would indeed be `model.export(format='engine', device=0)`. We appreciate your attention to detail and will make sure to update the documentation to reflect this. If you have any more feedback or questions, feel free to share! 😊🛠️

**Comment by digogogo** [comment]
how to slove this
Traceback (most recent call last):
  File "D:\yolov8\ultralytics-main\ultralytics\engine\model.py", line 418, in _smart_load
    return self.task_map[self.task][key]
  File "D:\python\miniconda\envs\yolov8\lib\site-packages\torch\nn\modules\module.py", line 1695, in __getattr__
    raise AttributeError(f"'{type(self).__name__}' object has no attribute '{name}'")
AttributeError: 'YOLO' object has no attribute 'task_map'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "D:\yolov8\ultralytics-main\coco-train.py", line 7, in <module>
    results = model.train(data='coco8.yaml', epochs=100, imgsz=640)
  File "D:\yolov8\ultralytics-main\ultralytics\engine\model.py", line 336, in train
    self.trainer = (trainer or self._smart_load('trainer'))(overrides=args, _callbacks=self.callbacks)
  File "D:\yolov8\ultralytics-main\ultralytics\engine\model.py", line 422, in _smart_load
    raise NotImplementedError(
NotImplementedError: WARNING  'YOLO' model does not support 'train' mode for 'detect' task yet.

**Reply by pderrenger** [reply]
@digogogo it appears you've encountered an error indicating that the 'train' mode for the 'detect' task is not yet supported by the 'YOLO' model object you're using. This could be due to an issue with the model configuration or an unsupported operation in the current context of the code.

To resolve this, please ensure that you are using the correct model configuration file and that the 'train' mode is indeed supported for the 'detect' task in the version of YOLOv8 you are using. If you're using a custom model configuration, double-check that it is set up correctly.

If the issue persists, consider reviewing the documentation for any updates on supported tasks and modes, or reach out to the community for further assistance. Remember to keep your codebase updated with the latest version of YOLOv8, as updates may include fixes and new features that could resolve your issue. 🛠️
